src/current/v23.2/changefeed-messages.md

```diff
@@ -191,70 +191,134 @@ CREATE CHANGEFEED FOR TABLE rides INTO 'external://kafka' WITH diff, envelope=wr
+    {{site.data.alerts.callout_info}}
+    Depending on the workload, you can use resolved timestamp notifications on every Kafka partition to provide strong ordering and global consistency guarantees by buffering records in between timestamp closures. Use the `resolved` timestamp to see every row that changed at a certain time.
+    {{site.data.alerts.end}}
 
-    For example:
+#### Define a key column
 
-    ~~~json
-    {"__crdb__": {"updated": "1532377312562986715.0000000000"}, "id": 1, "name": "Petee H"}
-    {"__crdb__": {"updated": "1532377306108205142.0000000000"}, "id": 2, "name": "Carl"}
-    {"__crdb__": {"updated": "1532377358501715562.0000000000"}, "id": 3, "name": "Ernie"}
-    {"__crdb__":{"resolved":"1532379887442299001.0000000000"}}
-    {"__crdb__":{"resolved":"1532379888444290910.0000000000"}}
-    {"__crdb__":{"resolved":"1532379889448662988.0000000000"}}
-    ...
-    {"__crdb__":{"resolved":"1532379922512859361.0000000000"}}
-    {"__crdb__": {"updated": "1532379923319195777.0000000000"}, "id": 4, "name": "Lucky"}
-    ~~~
+Typically, changefeeds that emit to Kafka sinks shard rows between Kafka partition using the row's primary key. The primary key is computed into a hash, which remains the same and ensures a row will always emit to the same Kafka partition.
+
+In some cases, you may want to specify another column in a table as the key by using the [`key_column`]({% link {{ page.version.version }}/create-changefeed.md %}#key-column) option, which will determine the partition your messages will emit to. However, if you implement `key_column` with a changefeed, consider that other columns may have arbitrary values that change. As a result, the same row (i.e., by primary key) may emit to any partition at the sink based upon the column value. A changefeed with a `key_column` specified will still maintain per-key and at-least-once delivery guarantees.
 
-- With duplicates removed, an individual row is emitted in the same order as the transactions that updated it. However, this is not true for updates to two different rows, even two rows in the same table.
+To confirm that messages may emit the same row to different partitions when an arbitrary column is used, you must include the `unordered` option:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED FOR TABLE employees INTO 'external://kafka-sink'
+    WITH key_column='office', unordered;
+~~~
+
+### At-least-once delivery
+
+Changefeeds also provide an _at-least-once delivery guarantee_, which means that each version of a row will be emitted once. Under some infrequent conditions a changefeed will emit duplicate messages. This happens when the changefeed was not able to emit all messages before reaching a checkpoint. As a result, it will re-emit all messages starting from the previous checkpoint to ensure every message is delivered at least once.
```

Comment:
> which means that each version of a row will be emitted once

We should tack on the end here, "..., but could be emitted more than once.". Otherwise it reads as if this is exactly once.

Reply:
Done!